src/service/webhook

```diff
@@ -1,250 +1,279 @@
+                'project': self.project,
+                'package': self.package,
+                #'repourl': self.args.repourl,
+                #'branch': self.args.branch,
+            }
+            self._debug('Fetching old whm')
+            self._debug(params)
+            r = requests.get(
+                '%s/webhookmappings/' % self.WEBHOOK_URL,
+                auth=self.auth,
+                params=params,
+            )
+            r.raise_for_status()
+            response = r.json()
+            self._debug(response)
+            if response['count'] > 1:
+                raise Exception(
+                    'More than one webhook mapping found for %s' % (params,)
+                )
+            elif response['count'] == 0:
+                self._debug('No old whm found')
+                self._old_whm = None
+            else:
+                self._old_whm = response['results'][0]
+                self._debug('Got old whm')
+                self._debug(self._old_whm)
+        return self._old_whm
+
+    def update_old_whm(self):
+        self._debug("Updating old whm")
+        r = requests.patch(
+            '%s/webhookmappings/%s/' % (self.WEBHOOK_URL, self.old_whm['id']),
+            auth=self.auth,
+            json=self.new_whm,
+        )
+        r.raise_for_status()
+        result = r.json()
+        self._debug(result)
+        return result
+
+    def create_new_whm(self):
+        self._debug("Creating new whm")
+        r = requests.post(
+            '%s/webhookmappings/' % self.WEBHOOK_URL,
+            auth=self.auth,
+            json=self.new_whm,
+        )
+        r.raise_for_status()
+        result = r.json()
+        self._debug(result)
+        return result
+
+    def trigger_whm(self, whm):
+        self._debug("Triggering whm %s" % whm['id'])
+        r = requests.put(
+            '%s/webhookmappings/%s/trigger/' % (self.WEBHOOK_URL, whm['id']),
+            auth=self.auth,
+        )
+        r.raise_for_status()
+        result = r.json()
+        self._debug(result)
+        return result
+
+    @property
+    def whm_needs_update(self):
+        keys = [
+            "repourl",
+            "branch",
+            "project",
+            "package",
+            "token",
+            "debian",
+            "dumb",
+            "notify",
+            "build",
+            "comment",
+        ]
+        errors = []
+
+        for key in keys:
+            old = self.old_whm.get(key, None)
+            new = self.new_whm.get(key, None)
+            if old != new:
+                errors.append(
+                    "%s: differs (%s != %s)" %
+                    (key, old, new)
+                )
+
+        if len(errors) > 0:
+            self._debug("Don't match:")
+            for e in errors:
+                self._debug(" %s" % e)
+            return True
+        return False
+
+    def _debug(self, msg):
+        if self.DEBUG:
+            print(msg)
+
+    def main(self):
```

Comment:
This logic seems wrong.
It should be:
if self.old_whm:
  if self.whm_needs_update:
    self.update_old_whm
    self.trigger
  else:
    pass # important to have a no-op here I think
else:
  self.create_new
  self.trigger